ipe_process_orchestrator/assignment_flow.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import sys
+import json
+from typing import Any, Dict, List, NoReturn, Optional, Union
+from requests.models import Response
+from constants import (
+    CANVAS_URL_BEGIN, ASSIGNMENT_GROUP_NAME, ASSIGNMENT_NAME)
+from api_handler.api_calls import APIHandler
+from ipe_process_orchestrator.api_helper import response_none_check
+
+logger = logging.getLogger(__name__)
+
+
+class IPEAssignmentFlow:
+
+    def __init__(self, api_handler, course_id, rubric_id) -> None:
+        self.rubric_id: str = rubric_id
+        self.course_id: int = course_id
+        self.api_handler: APIHandler = api_handler
+
+    def _look_up_ipe_assignment(self) -> Union[List[int], int]:
+        """
+        This is the use case for copied course that might have the same IPE assignment/group names as the original course.
+        Just look up assignment group by name and then look up the assignment.
+        Will be fetching 100 records only and it is highly unlikely that more than 100 assignment groups will there.
+        the assignment groups will also bring in assignment names to check and compare.
+        """
+        lookup_ag_payload: Dict = {'include[]': 'assignments', 'per_page': 100}
+        lookup_ag_url: str = f'{CANVAS_URL_BEGIN}/courses/{self.course_id}/assignment_groups'
+        lookup_ag_resp: Optional[Response] = self.api_handler.api_call_with_retries(
+            lookup_ag_url, 'GET', lookup_ag_payload)
+
+        err_msg: str = f"Error looking up assignment group '{ASSIGNMENT_GROUP_NAME}' with assignment name '{ASSIGNMENT_NAME}' for course failed {self.course_id}"
```

Comment:
a slight change from 

`for course failed {self.course_id}
`
to 
`for course {self.course_id} failed`